src/include/daos_prop.h

```diff
@@ -67,17 +67,68 @@ enum daos_pool_props {
 	 * The pool svc rank list.
 	 */
 	DAOS_PROP_PO_SVC_LIST,
+	/**
+	 * Pool cell size.
+	 */
 	DAOS_PROP_PO_EC_CELL_SZ,
 	/**
 	 * Media selection policy
 	 */
 	DAOS_PROP_PO_POLICY,
+	/**
+	 * Pool redundancy factor.
+	 */
+	DAOS_PROP_PO_REDUN_FAC,
+	/**
+	 * The pool performance domain affinity level of EC object.
+	 */
+	DAOS_PROP_PO_EC_PDA,
+	/**
+	 * The pool performance domain affinity level
+	 * of replicated object.
+	 */
+	DAOS_PROP_PO_RP_PDA,
 	DAOS_PROP_PO_MAX,
 };
 
 #define DAOS_PROP_PO_EC_CELL_SZ_MIN	(1UL << 10)
 #define DAOS_PROP_PO_EC_CELL_SZ_MAX	(1UL << 30)
 
+#define DAOS_PROP_PO_REDUN_FAC_MAX	4
+#define DAOS_RPOP_PO_REDUN_FAC_DEFAULT	0
+
+static inline bool
+daos_rf_is_valid(unsigned long long rf)
+{
+	return rf <= DAOS_PROP_PO_REDUN_FAC_MAX;
+}
+
+/**
+ * the placement algorithm should place two-way and three-way
+ * replication object within a PD; for those object classes with
+ * more than 3 replicas, DAOS will place three replicas within a PD
+ * and switch to another PD.
+ */
+#define DAOS_PROP_PO_RP_PDA_DEFAULT	3
+/**
+ * the placement algorithm always tries to scatter shards of EC
+ * object to different PDs.
+ */
+#define DAOS_PROP_PO_EC_PDA_DEFAULT	1
+
+/**
+ * the placement algorithm should place two-way and three-way
+ * replication object within a PD; for those object classes with
+ * more than 3 replicas, DAOS will place three replicas within a PD
+ * and switch to another PD.
+ */
+#define DAOS_PROP_PO_RP_PDA_DEFAULT	3
+/**
+ * the placement algorithm always tries to scatter shards of EC
+ * object to different PDs.
+ */
+#define DAOS_PROP_PO_EC_PDA_DEFAULT	1
```

Comment:
Looks like each of these values is defined twice. Is it a copy/paste error or did you intend one of the pairs to be DAOS_PROP_CO_... instead?

Reply:
removed duplicate